File: 1422-maximum-score-after-splitting-a-string/1422-maximum-score-after-splitting-a-string.cpp
class Solution {
public:
    int maxScore(string s) {
        int one = 0;
        for(char ch : s){
            if (ch == '1'){
                one++;
            }
        }

        int answer = 0;
        int zero = 0;
        for (int i = 0; i < s.length() - 1; i++){
            if(s[i] == '1'){
                one--;
            } else {
                zero++;
            }

            answer = max(answer, one + zero);
        }

        return answer;
    }
};

File: 1422-maximum-score-after-splitting-a-string/1422-maximum-score-after-splitting-a-string.py
class Solution:
    def maxScore(self, s: str) -> int:
        right = 0
        for ch in s:
            if ch == '1':
                right += 1

        answer = 0
        left = 0
        for ch in s[:len(s) - 1]:
            if ch == '0':
                left += 1
            else:
                right -= 1
            answer = max(answer, left + right)
        
        return answer